### PR TITLE
Upgrade gradle, migrate to kts, patch application slightly

### PR DESCRIPTION
**Pull request by Doomsdayrs**
#### Description
This application has been left unmaintained for a long time. To improve compatibility going forward, 
1. I migrated the gradle scripts to kotlin scripts.
2. Migrated application to androidx and removed support libraries
3. Replaced deprecated kotlin-android-extensions with viewbinding instead

Due to the nature of this type of PR, A great deal had to be changed, thankfully it wasn't too destructive.

Some notes, 
- I marked off some TODOs about deprecated apis and how some parts of the application will cause errors with android 12.
- Fabric is deprecated, and for crash reporting I suggest using ACRA instead of google services.
- For the future, please attempt to migrate to an MVVM architecture, the code base is messy.
- The android tests have been commented out, as it was out of the scope of this PR to repair them.

#### Type of PR
- [x] Refactor of code

#### Testing
Ensure that application builds properly.

### CODE REVIEW

**Comment by skraynick** [comment]
Will look at in the morning. Thanks.

File: mobile/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		named("release") {
+			isMinifyEnabled = false
+			proguardFiles(getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro")
+		}
+	}
+
+	buildFeatures {
+		dataBinding = true
+		viewBinding = true
+		// for view binding :
```
Comment (skraynick) [inline mobile/build.gradle.kts]: Let's remove commented code.

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		maven("https://maven.fabric.io/public")
+		maven("https://maven.google.com")
+	}
+	dependencies {
+		classpath("com.android.tools.build:gradle:7.0.3")
+		classpath(kotlin("gradle-plugin", version = "1.5.31"))
+		classpath(kotlin("serialization", version = "1.5.31"))
+		classpath("com.google.gms:google-services:4.3.10")
+		classpath("com.google.firebase:firebase-crashlytics-gradle:2.8.0")
+		// NOTE: Do not place your application dependencies here; they belong
```
Comment (skraynick) [inline build.gradle.kts]: Cleanup Comments.

File: mobile/build.gradle.kts (hunk 1 above)
Comment (Doomsdayrs) [reply mobile/build.gradle.kts]: Didn't even notice the redundancy, Removed

File: build.gradle.kts (hunk 2 above)
Comment (Doomsdayrs) [reply build.gradle.kts]: This seems to be a comment left by android studio when generating the original files, removed anyways

File: mobile/src/main/java/mycroft/ai/MainActivity.kt [hunk 3]
```diff
@@ -38,466 +38,475 @@ import androidx.appcompat.app.AppCompatActivity
...
+		llm.orientation = LinearLayoutManager.VERTICAL
+		with(binding.contentMain.cardList) {
+			setHasFixedSize(true)
+			layoutManager = llm
+			adapter = mycroftAdapter
+		}
+
+		registerReceivers()
+
+		// start the discovery activity (testing only)
```
Comment (skraynick) [inline mobile/src/main/java/mycroft/ai/MainActivity.kt:151]: Probably shouldn't leave testing commented code in.

File: mobile/src/main/java/mycroft/ai/MainActivity.kt [hunk 4]
```diff
@@ -38,466 +38,475 @@ import androidx.appcompat.app.AppCompatActivity
...
+				null
+			}
+		} else {
+			null
+		}
+	}
+
+	fun sendMessage(msg: String) {
+		// let's keep it simple eh?
+		//final String json = "{\"message_type\":\"recognizer_loop:utterance\", \"context\": null, \"metadata\": {\"utterances\": [\"" + msg + "\"]}}";
```
Comment (skraynick) [inline mobile/src/main/java/mycroft/ai/MainActivity.kt]: I dislike left over comments. I know not you, but let's clean them up..

**Review by skraynick — COMMENTED** [review]
Fix the one issue, and I will merge. Best. S

File: mobile/src/main/java/mycroft/ai/MainActivity.kt (hunk 4 above)
Comment (Doomsdayrs) [reply mobile/src/main/java/mycroft/ai/MainActivity.kt]: Boom

**Pull request merged**